src/volttron/client/commands/install_agents.py

```diff
@@ -470,29 +440,35 @@ def send_zmq():
             rmq_response_topic,
         )
     elif server.core.messagebus == "zmq":
-        _log.debug(f"calling install_agent on {peer} using channel {channel.name}")
-        task = gevent.spawn(send_zmq)
+        if wheel_install:
+            _log.debug(f"calling install_agent on {peer} using channel {channel.name}")
+            task = gevent.spawn(send_zmq)
+            agent_package = os.path.basename(path)
+            channel_name = channel.name
+        else:
+            agent_package = agent
+            channel_name = None
         result = server.vip.rpc.call(
             peer,
             "install_agent",
-            os.path.basename(path),
-            channel.name,
+            agent_package,
+            channel_name,
             vip_identity,
             publickey,
             secretkey,
             force,
             agent_config,
         )
-
     else:
         raise ValueError("Unknown messagebus detected!")
-
-    result.rawlink(lambda glt: task.kill(block=False))
-    # Allows larger files to be sent across the message bus without
-    # raising an error.
-    gevent.wait([result], timeout=300)
-    _log.debug("Completed sending of agent across.")
-    _log.debug(f"After wait result is {result}")
+    try:
+        if wheel_install:
+            result.rawlink(lambda glt: task.kill(block=False))
+            # Allows larger files to be sent across the message bus without
+            # raising an error.
+        gevent.wait([result], timeout=300)
```

Comment:
Would the nested if/else logic above ever lead to `result` being None?

Reply:
it shouldn't. it should either return agent uuid or throw exception